File: fr/eni/jenkins/GestionDeListe.java
package fr.eni.jenkins;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class GestionDeListe {

	public static void main(String[] args) {
		// Listes : methodes et indicage.
		// fichier : GestionDeListe.java
		// auteur : ENI Ecole
		// programme principal -----------------------------------------------
		List<Integer> liste = new ArrayList<>();
		liste.add(17);
		liste.add(38);
		liste.add(10);
		liste.add(25);
		liste.add(72);
		System.out.println("----------------- Liste initiale -----------------");
		System.out.println(liste);
		System.out.println("\n");

		System.out.println("-------- Liste triée par ordre croissant --------");
		Collections.sort(liste);
		System.out.println(liste);
		System.out.println("\n");

		System.out.println("------------- Ajout de la valeur 12 -------------");
		liste.add(12);
		Collections.sort(liste);
		System.out.println(liste);
		System.out.println("\n");

		System.out.println("------------------ Indice de 17 ------------------");
		System.out.println(liste.indexOf(17));
		System.out.println("\n");

		System.out.println("----------------- Retrait de 38 -----------------");
		Integer val = 38;
		liste.remove(val);
		System.out.println(liste);
		System.out.println("\n");

		System.out.println("------- Sous-liste du 2ème au 3ème élément -------");

		System.out.println(liste.subList(1, 3));
	}
}

File: IC_Jenkins_TP01_DOTNET/GestionDeListe.cs
using System;
using System.Collections.Generic;

namespace IC_Jenkins_TP01_C_
{
    public class GestionDeListe
    {
        public static void Affiche(List<int> liste){
            Console.Write("[");
            int i;
            for (i = 0; i < liste.Count - 1; i++)
            {
                Console.Write(liste[i] + ", ");
            }
            Console.Write(liste[i] + "]\n\n");
        }
        public static void Main()
        {
            // Listes : methodes et indicage.
		// fichier : GestionDeListe.java
		// auteur : ENI Ecole
		// programme principal -----------------------------------------------
		List<int> liste = new List<int>();
        liste.Add(17);
		liste.Add(38);
		liste.Add(10);
		liste.Add(25);
		liste.Add(72);
		Console.WriteLine("----------------- Liste initiale -----------------");
		Affiche(liste);

		Console.WriteLine("-------- Liste triée par ordre croissant --------");
		liste.Sort();
		Affiche(liste);

		Console.WriteLine("------------- Ajout de la valeur 12 -------------");
		liste.Add(12);
		liste.Sort();
		Affiche(liste);

		Console.WriteLine("------------------ Indice de 17 ------------------");
		Console.WriteLine(liste.IndexOf(17));
		Console.WriteLine("\n");

		Console.WriteLine("----------------- Retrait de 38 -----------------");
		liste.Remove(38);
		Affiche(liste);

		Console.WriteLine("------- Sous-liste du 2ème au 3ème élément -------");
		List<int> subList = liste.GetRange(1,2);
        Affiche(subList);
        }
    }
}